src/batch/src/executor/expand.rs

```diff
@@ -97,6 +95,35 @@ impl ExpandExecutor {
     }
 }
 
+/// Expand the `columns` according to `column_subsets`.
+///
+/// This is a helper function for Expand operator.
+#[try_stream(boxed, ok = Vec<ArrayRef>, error = ArrayError)]
+async fn expand_columns(
```

Comment:
Can we use `std::iter::from_generator` here?

https://doc.rust-lang.org/stable/std/iter/fn.from_generator.html

Reply:
It seems we can. Let me have a try.
This code was simply moved from the common crate and I didn't change it.